### PR TITLE
Always auto name and create new workspaces from oldDot

### PR DESCRIPTION
**Pull request by timszot**
@TomatoToaster pinging you for review since we discussed this PR.

### Details
Removes the `workspace/new` page completely since we always want to auto name and create workspaces for N6.
[Slack thread](https://expensify.slack.com/archives/C020EPP9B9A/p1628547819480000) for more context, but in N6 we always want to auto name and create the workspace for the user, so the `workspace/new` page is no longer needed.

Also removed the success growl per [this](https://github.com/Expensify/Expensify/issues/173338) issue.

### Fixed Issues

$ https://github.com/Expensify/Expensify/issues/173349
$ https://github.com/Expensify/Expensify/issues/173338

### Tests
Tested through oldDot Inbox flow.
1. Log into an account on oldDot and show all hidden inbox cards
2. Find the inbox card below and select only `Get Expensify Card` then click `Submit`
image
3. Verify you are taken to `/workspace/<policyId>/card`
image

### QA Steps
Same as tests

### Tested On

- [x] Web
- [ ] Mobile Web
- [ ] Desktop
- [ ] iOS
- [ ] Android

#### Web
https://user-images.githubusercontent.com/3387421/128938529-9ca25e13-5942-4c79-8bd6-1807b4693b1e.mov

### CODE REVIEW

File: src/languages/en.js
Comment (timszot) [inline src/languages/en.js:603]: Verifying the ES translation for this still, will take off `HOLD` once confirmed.

**Review by TomatoToaster — DISMISSED** [review]
This LGTM pending the spanish translation. Feel free to merge when that's approved or this goes off hold.

**Comment by timszot** [comment]
Sorry Bondy for the pings, was trying to get pullerbear to pick someone else...

**Comment by timszot** [comment]
Ok @TomatoToaster `HOLD` removed and ready for one more review before it can be merged.

File: src/Expensify.js
Comment (bondydaa) [inline src/Expensify.js:119]: NAB but not a super useful comment, should explain why we are doing something not what we are doing.

File: src/pages/ValidateLogin2FANewWorkspacePage.js
Comment (bondydaa) [inline src/pages/ValidateLogin2FANewWorkspacePage.js:74]: NAB same as other comment, should explain why not what.

File: src/pages/ValidateLoginNewWorkspacePage.js
Comment (bondydaa) [inline src/pages/ValidateLoginNewWorkspacePage.js:50]: NAB same, explain why not what

**Review by bondydaa — DISMISSED** [review]
just a couple of NAB otherwise looks good to my newbie-app-eyes

**Review by bondydaa — APPROVED** [review]
👍

**Comment by timszot** [comment]
Self merging since tests are passing and this is approved.

**Pull request merged**
